pkg/splunk/enterprise/configuration.go

```diff
@@ -223,10 +225,26 @@ func setVolumeDefaults(spec *enterprisev1.CommonSplunkSpec) {
 	}
 }
 
+func setServiceTemplateDefaults(spec *enterprisev1.CommonSplunkSpec) {
+	if spec.CommonSpec.ServiceTemplate.Spec.Ports != nil {
```

Comment:
seems like we should also check for ServiceTemplate != nil and ServiceTemplate.Spec != nil

Reply:
These are inner structs.